### ISSUE TITLE
Options passed in "Job Reference" step are not being replaced by empty strings

### ISSUE THREAD

**Issue by jacekkow** [rundeck/rundeck#1272]
This is a special case of #352

In latest Rundeck (2.5.1) I have created a job (Instance) which references another one (Template):
```yml
- id: 3b04bd49-dc72-4f18-9012-ee30165ae482
  project: Test
  loglevel: DEBUG
  sequence:
    keepgoing: false
    strategy: node-first
    commands:
    - script: |
        #!/bin/bash
        
        echo $*
      args: ${option.required} ${option.optional}
  description: ''
  name: Template
  uuid: 3b04bd49-dc72-4f18-9012-ee30165ae482
  options:
    optional:
      regex: '[a-zA-Z]*'
    required:
      required: true
      regex: '[a-zA-Z]*'
```
```yml
- id: 6aba8811-3496-414f-ac02-c78704e1bbae
  project: Test
  loglevel: DEBUG
  sequence:
    keepgoing: false
    strategy: node-first
    commands:
    - jobref:
        group: ''
        name: Template
        args: -optional ${option.optional} -required ${option.required}
        nodeStep: 'true'
  description: ''
  name: Instance
  uuid: 6aba8811-3496-414f-ac02-c78704e1bbae
  options:
    optional:
      regex: '[a-zA-Z]*'
    required:
      required: true
      regex: '[a-zA-Z]*'
```

Comment in this pull request https://github.com/rundeck/rundeck/pull/440 suggests that when optional argument is not present, references to one will be replaced by blank string. It is true when "Template" is executed, but when I try to run "Instance" without entering value for option "optional", this error is shown:

```
[workflow] Begin execution: rundeck-workflow-node-first context: null
preparing for sequential execution on 1 nodes
Executing command on node: localhost, NodeEntryImpl{tags=[], attributes={tags=, osFamily=unix, username=rundeck, osVersion=3.16.0-4-amd64, osArch=amd64, description=Rundeck server node, hostname=localhost, nodename=localhost, osName=Linux}, project='null'}
NodeSet: MultiNodeSelector{nodenames=[localhost]}
Workflow: com.dtolabs.rundeck.core.execution.workflow.StepFirstWorkflowStrategy$stepFirstWrapper@aea29ab2
data context:  {node={tags=, username=rundeck, os-version=3.16.0-4-amd64, description=Rundeck server node, name=localhost, os-arch=amd64, hostname=localhost, os-name=Linux, os-family=unix}, job={loglevel=DEBUG, wasRetry=false, url=http://localhost:4440/project/Test/execution/follow/37, id=6aba8811-3496-414f-ac02-c78704e1bbae, project=Test, username=s4a00099, retryAttempt=0, user.name=s4a00099, name=Instance, serverUUID=null, group=null, execid=37, serverUrl=http://localhost:4440/project}, option={required=Required}}
[workflow] Begin step: 1,NodeDispatch
1: Workflow step executing: com.dtolabs.rundeck.execution.ExecutionItemFactory$5@33206795
preparing for sequential execution on 1 nodes
Executing command on node: localhost, NodeEntryImpl{tags=[], attributes={tags=, osFamily=unix, username=rundeck, osVersion=3.16.0-4-amd64, osArch=amd64, description=Rundeck server node, hostname=localhost, nodename=localhost, osName=Linux}, project='null'}
[workflow] beginExecuteNodeStep(localhost): NodeDispatch: com.dtolabs.rundeck.execution.ExecutionItemFactory$5@33206795
Option input was not valid for [Template]: Option 'optional' doesn't match regular expression [a-zA-Z]*, value: ${option.optional}
[workflow] finishExecuteNodeStep(localhost): NodeDispatch: InvalidOptions: Invalid options: [optional]
1: Workflow step finished, result: Dispatch failed on 1 nodes: [localhost: InvalidOptions: Invalid options: [optional]]
[workflow] Finish step: 1,NodeDispatch
[workflow] Finish execution:  rundeck-workflow-node-first: [Workflow result: , step failures: {1=Dispatch failed on 1 nodes: [localhost: InvalidOptions: Invalid options: [optional]]}, Node failures: {localhost=[InvalidOptions: Invalid options: [optional]]}, flow control: Continue, status: failed]
[Workflow result: , step failures: {1=Dispatch failed on 1 nodes: [localhost: InvalidOptions: Invalid options: [optional]]}, Node failures: {localhost=[InvalidOptions: Invalid options: [optional]]}, flow control: Continue, status: failed]
Execution failed: 37: [Workflow result: , step failures: {1=Dispatch failed on 1 nodes: [localhost: InvalidOptions: Invalid options: [optional]]}, Node failures: {localhost=[InvalidOptions: Invalid options: [optional]]}, flow control: Continue, status: failed]
```

Relevant part:
**Option input was not valid for [Template]: Option 'optional' doesn't match regular expression [a-zA-Z]*, value: ${option.optional}**

**Comment by richiereynolds** [comment]
+1 this might be a dup, I think I raised it before.

**Comment by richiereynolds** [comment]
Yep - https://github.com/rundeck/rundeck/issues/992

**Comment by jacekkow** [comment]
Closing as a duplicate.

**Comment by ScottG489** [comment]
I can still reproduce this issue on version **2.6.0-1**

Steps to reproduce:
1. Create new job
2. Add a option, leaving all properties of it at their defaults (not required, no default value, etc)
3. Add a Local Command step which echo's the option. In my case I called my option '**field**':
`echo ${option.field}`
4. Finish creating the job and navigate to it to run it
5. Run the job without entering anything into the text box

The output will be `${option.field}` literally

Not sure if this is a regression in functionality between the version it was allegedly fixed in (**2.5.2**) and my current version (**2.6.0-1**) but it's definitely still broken. I can create a new issue if that would be more appropriate.

Although this is the issue that was referenced as being fixed in your [release notes](http://rundeck.org/news/2015/07/07/rundeck-2.5.2.html#issues) I also posted this in response to #992 as well since it is the issue that was referenced as being fixed by commit [4dd190e](https://github.com/rundeck/rundeck/commit/4dd190e17a590f5730e782f646c4b729d2f5ca96)
